src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestResolversIdentitiesDataStore.java

```diff
@@ -0,0 +1,129 @@
+package com.minecolonies.coremod.colony.requestsystem.data;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.colony.requestsystem.data.IRequestResolverIdentitiesDataStore;
+import com.minecolonies.api.colony.requestsystem.factory.FactoryVoidInput;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolver;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.TypeConstants;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+/**
+ * A default implementation of the {@link IRequestResolverIdentitiesDataStore} interface.
+ */
+public class StandardRequestResolversIdentitiesDataStore implements IRequestResolverIdentitiesDataStore
+{
+    private IToken<?>                                   id;
+    private final BiMap<IToken<?>, IRequestResolver<?>> map;
+
+    public StandardRequestResolversIdentitiesDataStore(
+      final IToken<?> id,
+      final BiMap<IToken<?>, IRequestResolver<?>> map) {
+        this.id = id;
+        this.map = map;
+    }
+
+    public StandardRequestResolversIdentitiesDataStore()
+    {
+        this.id = StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN);
+        this.map = HashBiMap.create();
+    }
+
+    @Override
+    public BiMap<IToken<?>, IRequestResolver<?>> getIdentities()
+    {
+        return map;
+    }
+
+    @Override
+    public IToken<?> getId()
+    {
+        return id;
+    }
+
+    @Override
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestResolversIdentitiesDataStore>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)